ambari-server/src/main/java/org/apache/ambari/server/controller/internal/HostResourceProvider.java

```diff
@@ -238,6 +242,45 @@ public Void invoke() throws AmbariException, AuthorizationException {
   @Override
   protected Set<Resource> getResourcesAuthorized(Request request, Predicate predicate)
       throws SystemException, UnsupportedPropertyException, NoSuchResourceException, NoSuchParentResourceException {
+    if (request.getPropertyIds().contains(HOST_SUMMARY_PROPERTY_ID)) {
+      return getHostSummaryResource(request, predicate);
```

Comment:
Can you please document the response structure of host summary in `HostResponse` or a new interface using Swagger annotations?

Reply:
There are two issues here: 1. REST API does not allow two interfaces which have the same path but different parameters. So in this case /api/v1/hosts and /api/v1/hosts?format=summary are regarded as the same interface, we cannot create another API with only different paramter; 2. Swagger does not support one method with multiple endpoints, that is, /api/v1/hosts and /api/v1/hosts?format=query return totally differnt responses, and so we cannot use HostResponse to summary structure of host summary in HostResponse. So unless I use a different path such as /api/v1/hosts/summary, I cannot use swagger annotation